app/controllers/RaceGroupController.java

```diff
@@ -0,0 +1,112 @@
+package controllers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import models.RaceGroup;
+import models.enums.RaceGroupType;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import play.mvc.BodyParser;
+import play.mvc.Controller;
+import play.mvc.Result;
+import repository.interfaces.RaceGroupRepository;
+
+import javax.inject.Inject;
+import java.sql.Timestamp;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import static play.libs.Json.toJson;
+
+public class RaceGroupController extends Controller {
+    private final RaceGroupRepository raceGroupRepository;
+
+    @Inject
+    public RaceGroupController(RaceGroupRepository raceGroupRepository) {
+        this.raceGroupRepository = raceGroupRepository;
+    }
+
+    public CompletionStage<Result> getAllRaceGroups() {
+        return raceGroupRepository.getAllRaceGroups().thenApplyAsync(raceGroups -> ok(toJson(raceGroups.collect(Collectors.toList())))).exceptionally(ex -> {
+            Result res;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)